src/Smalot/PdfParser/Font.php

```diff
@@ -134,9 +134,16 @@ public function translateChar(string $char, bool $use_default = true)
 
     /**
      * Convert unicode character code to "utf-8" encoded string.
+     *
+     * @param int|float $code Unicode character code. Will be casted to int internally!
      */
-    public static function uchr(int $code): string
+    public static function uchr($code): string
     {
+        // note:
+        // $code was typed as int before, but changed in https://github.com/smalot/pdfparser/pull/623
+        // because in some cases uchr was called with a float instead of an integer.
+        $code = (int) $code;
```

Comment:
I suppose since there's no restriction on the types the function will accept (`uchr($code)`) then casting it to (int) makes sense. It could be a string, or anything else, even though likely not. A (float) will stay a (float) here, if it's too big, but it doesn't make any difference. :)

Reply:
Just for reference: https://www.php.net/manual/en/language.types.integer.php says:

> If PHP encounters a number beyond the bounds of the int type, it will be interpreted as a float instead. Also, an operation which results in a number beyond the bounds of the int type will return a float instead.